dds/DCPS/transport/shmem/ShmemTransport.cpp

```diff
@@ -260,7 +281,9 @@ ShmemTransport::ReadTask::svc()
     if (stopped_.value()) {
       return 0;
     }
-    outer_->read_from_links();
+    while (!(outer_->read_from_links() || stopped_.value())) {
+      ACE_OS::sleep(1);
```

Comment:
I've kept this sleep in so far, but I'm not sure if it's still necessary or not.